### DISCUSSION TITLE
What is the proper way to edit react-query's cached data?

### DISCUSSION THREAD

**Discussion by branlok** [TanStack/query#2098] [Q&A] [answered]
I'm using useInifiniteQuery fetching a long list of posts. I've set up a liked counter and button. I don't want to refetch the entire list of posts to revalidate the server's truth. I was wondering if there is a way to edit the cache data such that i can manipulate "number of likes". Much appriciated.

**Comment by TkDodo** [comment, marked as answer]
`queryClient.setQueryData` is for that use-case. You can just call it after the mutation to update the cache. It's described here: https://react-query.tanstack.com/guides/updates-from-mutation-responses

be aware that infinite queries have a different structure (pages and pageParams) that you need to keep with your direct cache maniuplation.

**Reply by branlok** [reply]
thank you for providing the link!
